src/borg/repository.py

```diff
@@ -96,6 +96,9 @@ class InvalidRepository(Error):
     class CheckNeeded(ErrorWithTraceback):
         """Inconsistency detected. Please run "borg check {}"."""
 
+    class RepoBeyondRepair(ErrorWithTraceback):
+        """Heavily damaged repository {} detected - maybe this is beyond being repairable."""
+
```

Comment:
hmm, considering RPC, maybe adding a new exception is not the best way to deal with this.

i'll check if it can be done otherwise...